src/asmcnc/keyboard/custom_keyboard.py

```diff
@@ -0,0 +1,108 @@
+# -*- coding: utf-8 -*-
+import sys
+from kivy.core.window import Window
+from kivy.uix.vkeyboard import VKeyboard
+
+try:
+    import hgtk
+except:
+    pass
+
+
+class Keyboard(VKeyboard):
+    def __init__(self, text_inputs, **kwargs):
+        super(Keyboard, self).__init__(**kwargs)
+
+        self.l = kwargs['localization']
+
+        # For python 2 encoding issues
+        reload(sys)
+        sys.setdefaultencoding('utf-8')
+
+        self.text_instance = None
+
+        self.kr_layout = "./asmcnc/keyboard/layouts/kr.json"
+        self.numeric_layout = "./asmcnc/keyboard/layouts/numeric.json"
+        self.qwerty_layout = "data/keyboards/qwerty.json"
+        self.qwertyKR_layout = "./asmcnc/keyboard/layouts/qwertyKR.json"
+
+        try:
+            if self.l.lang == "Korean (KO)":
+                self.font = self.l.korean_font
+                self.layout = self.kr_layout
+            else:
+                self.font_name = "data/fonts/DejaVuSans.ttf"
+                self.layout = self.qwerty_layout
+        except:
+            pass
+
+        self.do_translation = False
+        self.width = Window.width
+        self.height = int(Window.height / 2.1)
+        self.pos = (Window.width - self.width, 0)
+        self.on_key_up = self.key_up
+
+        self.setup_text_inputs(text_inputs)
+
+    def setup_text_inputs(self, text_inputs):
+        for text_input in text_inputs:
+            try:
+                text_input.keyboard_mode = 'managed'
+                text_input.bind(focus=self.on_focus)
+            except:
+                print("setup failed")
```

Comment:
@VladPavelescu

Reply:
The debugging print isn't NEEDED but I do think we should have one just in case. A traceback would definitely be more useful. We can also print the actual `text_input` which will give the exact instance that failed.